pkg/install/install.go

```diff
@@ -0,0 +1,467 @@
+package install
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/projectcalico/libcalico-go/lib/names"
+	"github.com/prometheus/common/log"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"k8s.io/client-go/rest"
+)
+
+type config struct {
+	// Location on the host where CNI network configs are stored.
+	CNINetDir   string `envconfig:"CNI_NET_DIR" default:"/etc/cni/net.d"`
+	CNIConfName string `envconfig:"CNI_CONF_NAME"`
+
+	// Directory where we expect that TLS assets will be mounted into the calico/cni container.
+	TLSAssetsDir string `envconfig:"TLS_ASSETS_DIR"`
+
+	// SkipCNIBinaries is a comma-separated list of binaries. Each binary in the list
+	// will be skipped when installing to the host.
+	SkipCNIBinaries string `envconfig:"SKIP_CNI_BINARIES"`
+
+	// UpdateCNIBinaries controls whether or not to overwrite any binaries with the same name
+	// on the host.
+	UpdateCNIBinaries bool
+
+	// The CNI network configuration to install.
+	CNINetworkConfig     string `envconfig:"CNI_NETWORK_CONFIG"`
+	CNINetworkConfigFile string `envconfig:"CNI_NETWORK_CONFIG_FILE"`
+
+	ServiceAccountToken []byte
+}
+
+func (c config) skipBinary(binary string) bool {
+	return strings.Contains(c.SkipCNIBinaries, binary)
```

Comment:
Should really split by comma and then compare. Probably doesn't matter too much, but as it stands "my-binary" would match "not-my-binary"